core/client/fs/src/main/java/alluxio/client/file/cache/NoExceptionCacheManager.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.client.file.cache;
+
+import alluxio.metrics.MetricKey;
+import alluxio.metrics.MetricsSystem;
+
+import com.codahale.metrics.Counter;
+
+import java.nio.channels.ReadableByteChannel;
+
+import javax.annotation.Nullable;
+
+/**
+ * A wrapper class of CacheManager without throwing unchecked exceptions.
+ */
+public class NoExceptionCacheManager implements CacheManager {
```

Comment:
Should this catch throwable, and should we log when we catch?

Reply:
add log. but IMO  `Error` should be escalated as these are not something we should recover.